src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnectionString.cs

```diff
@@ -430,6 +486,52 @@ internal SqlConnectionString(string connectionString) : base(connectionString, G
                 throw ADP.InvalidConnectRetryIntervalValue();
             }
 
+            // Retry Logic
+            if ((_retryStrategy != "None") && (_retryStrategy != "FixedInterval") && (_retryStrategy != "Incremental") && (_retryStrategy != "ExponentialBackoff"))
+            {
+                throw ADP.InvalidRetryStrategy();
+            }
+
+            if ((_retryCount < 0) || (_retryCount > 60))
```

Comment:
Ciao Silvano, micro-optimization `(uint)_retryCount > 60` better code generation one less branch.

Reply:
Why is there a limit of 60 in the first place? I can see it making sense to retry a lot more. Maybe it is a good idea to retry once per second for one hour. That's 3600 times.

Reply:
If a failure last for like one hour stops to be "transient" and become a permanent one, so generally speaking even 60 times is more than it's effectively needed. Even more than that, a documented best practice for Azure SQL Database (https://docs.microsoft.com/en-us/azure/sql-database/sql-database-connectivity-issues#retry-logic-for-transient-errors) is to actually retry with at least a 5 sec interval minimum, otherwise that's going to be counterproductive. Usually, after a certain amount of retries (say 3-5) it's much better to return the underlying exception and let the application logic to deal with it.

Reply:
Is this feature only meant for Azure SQL?

I would caution against judging what's "best practice" so narrowly. Customer scenarios are very diverse.

If the only reason for that 60 retry limit is to force customers to use a "best practice" then I think that limit causes more harm than good. Let customers chose what's right for them.